db/main/migrate/20181116800000_add_constraints_to_multiple_tables.rb

```diff
@@ -0,0 +1,44 @@
+class AddConstraintsToMultipleTables < ActiveRecord::Migration[4.2]
+  def change
+    add_foreign_key :repositories, :builds, column: :current_build_id, on_delete: :nullify
+    add_foreign_key :repositories, :builds, column: :last_build_id, on_delete: :nullify
+
+    add_foreign_key :builds, :repositories, on_delete: :cascade
```

Comment:
I think adding `cascade` makes sense at the data level, but will this be confusing when used in conjunction with soft-deletes? We can probably overcome that by communicating that app developers need to handle that themselves (i.e. do soft-deletes not actual deletes), but it would be bad to accidentally delete public build records, for instance, by accidentally actually deleting a repository.

On the other hand, this seems super nice for those times where a user writes in wanting all of their data deleted. Just delete their repos and you're done.

Reply:
Oops, totally missed that. I don't think that's an issue. We don't really hard delete repositories and even if we do, I think it's worse to delete a repository and leave zombie builds behind, rather than delete the builds. Of course if it's done by accident, it will be harder to recover from than without these constraints, but in my opinion this is not something that should drive the database design. The alternative would be to nullify foreign keys, which will just leave unassigned builds lingering in the DB.